test/integration/esys-certifyX509.int.c

```diff
@@ -0,0 +1,243 @@
+                 .buffer = {0},
+             },
+        },
+    };
+
+    inSensitivePrimary.sensitive.userAuth = authValuePrimary;
+
+    TPM2B_PUBLIC inPublic = {
+        .size = 0,
+        .publicArea = {
+            .type = TPM2_ALG_RSA,
+            .nameAlg = TPM2_ALG_SHA256,
+            .objectAttributes = (
+                    TPMA_OBJECT_FIXEDTPM |
+                    TPMA_OBJECT_USERWITHAUTH |
+                    TPMA_OBJECT_RESTRICTED |
+                    TPMA_OBJECT_SIGN_ENCRYPT |
+#if defined TCTI_MSSIM
+                    TPMA_OBJECT_X509SIGN | /* This requires mssim >= v1628 Other simulators
+                                              that do not implement 1.59 spec return 0x000002e1
+                                              tpm:parameter(2):reserved bits not set to zero as required
+                                            */
+#endif
+                    TPMA_OBJECT_FIXEDTPM |
+                    TPMA_OBJECT_FIXEDPARENT |
+                    TPMA_OBJECT_SENSITIVEDATAORIGIN),
+            .authPolicy = {
+                    .size = 0,
+                    .buffer = {0},
+            },
+            .parameters.rsaDetail = {
+                .symmetric = {
+                    .algorithm = TPM2_ALG_NULL,
+                    .keyBits.aes = 0,
+                    .mode.aes = TPM2_ALG_NULL,
+                },
+                .scheme = {
+                     .scheme = TPM2_ALG_RSASSA,
+                     .details = { .rsassa = { .hashAlg = TPM2_ALG_SHA256 }},
+                },
+                .keyBits = 2048,
+                .exponent = 0,
+            },
+            .unique.rsa = {
+                .size = 0,
+                .buffer = {0},
+            },
+        },
+    };
+
+    TPM2B_AUTH authValue = {
+        .size = 0,
+        .buffer = {0}
+    };
+
+    TPM2B_DATA outsideInfo = {
+        .size = 0,
+        .buffer = {0},
+    };
+
+    TPML_PCR_SELECTION creationPCR = {
+        .count = 0,
+    };
+
+    TPMT_SIG_SCHEME inScheme = {
+        .scheme = TPM2_ALG_RSASSA,
+        .details.rsassa = TPM2_ALG_SHA256
+    };
+
+    LOG_INFO("\nRSA key will be created.");
+    r = Esys_TR_SetAuth(esys_context, ESYS_TR_RH_OWNER, &authValue);
+    goto_if_error(r, "Error: TR_SetAuth", error);
+
+    r = Esys_CreatePrimary(esys_context, ESYS_TR_RH_OWNER, ESYS_TR_PASSWORD,
+                           ESYS_TR_NONE, ESYS_TR_NONE, &inSensitivePrimary,
+                           &inPublic, &outsideInfo, &creationPCR,
+                           &signHandle, &outPublic, &creationData,
+                           &creationHash, &creationTicket);
+    goto_if_error(r, "Error esys create primary", error);
+
+    /* For the partial cert we need these fields:
+     * 1) Signature Algorithm Identifier (optional)
+       2) Issuer (mandatory)
+       3) Validity (mandatory)
+       4) Subject Name (mandatory)
+       5) Extensions (mandatory)
+       Data generated with IBM's tpmutils
```

Comment:
What was the exact command? Is their a way to generate this with openssl?

Reply:
I used this tool: https://sourceforge.net/p/ibmtpm20tss/tss/ci/master/tree/utils/certifyx509.c
Openssl will always create a "full cert", which will not work with TPM2_Certifyx509 command unless is it manually truncated and tweaked.

Reply:
Now we can use this tool: https://github.com/tpm2-software/tpm2-tools/pull/2151/files#diff-c0e10ea13892dafc56e08bdcc6b7d32e